src/webgpu/api/validation/queue/submit.spec.ts

```diff
@@ -53,10 +70,68 @@ g.test('command_buffer,duplicate_buffers')
     `
   )
   .fn(t => {
-    const encoder0 = t.device.createCommandEncoder();
-    const cb = encoder0.finish();
+    const cb = t.createCommandBuffer();
 
     t.expectValidationError(() => {
       t.device.queue.submit([cb, cb]);
     }, true);
   });
+
+g.test('command_buffer,submit_invalidates')
+  .desc(
+    `
+    Tests that calling submit invalidates the command buffers passed to it:
+    `
+  )
+  .fn(t => {
+    const cb = t.createCommandBuffer();
+
+    // Initial submit of a valid command buffer should pass.
+    t.device.queue.submit([cb]);
+
+    // Subsequent submits of the same command buffer should fail.
+    t.expectValidationError(() => {
+      t.device.queue.submit([cb]);
+    });
+  });
+
+g.test('command_buffer,invalid_submit_invalidates')
+  .desc(
+    `
+    Tests that calling submit invalidates all command buffers passed to it, even
+    if they're part of an invalid submit.
+    `
+  )
+  .fn(t => {
+    const cb = t.createCommandBuffer();
+    const cb_invalid = t.createCommandBuffer({ valid: false });
+
+    // Submit should fail because on of the command buffers is invalid
+    t.expectValidationError(() => {
+      t.device.queue.submit([cb, cb_invalid]);
+    });
+
+    // Subsequent submits of the previously valid command buffer should fail.
+    t.expectValidationError(() => {
+      t.device.queue.submit([cb]);
+    });
+  });
+
+g.test('command_buffer,submit_invalidate_order')
```

Comment:
Realizing now this is actually testing the same thing as the previous test. These two could be merged back, though no strong feeling.

Reply:
Merged them back again.